h11/_events.py

```diff
@@ -146,13 +87,31 @@ def _validate(self):
 
         validate(request_target_re, self.target, "Illegal target characters")
 
+    def __repr__(self):
+        return "{}(method={}, target={}, headers={}, http_version={})".format(
+            self.__class__.__name__,
+            self.method,
+            self.target,
+            self.headers,
+            self.http_version,
+        )
 
-class _ResponseBase(_EventBundle):
-    _fields = ["status_code", "headers", "http_version", "reason"]
-    _defaults = {"http_version": b"1.1", "reason": b""}
+    # Useful for tests
+    def __eq__(self, other):
+        if not isinstance(other, type(self)):
```

Comment:
Whilst this changes the logic of what is equal or not (from `self.__class__ == other.__class__`) to include subclasses and potentially raising if incompatible classes are compared (rather than returning False) I think it is fine as this is mostly (and likely only) used for tests.